library/src/main/java/com/yayandroid/locationmanager/providers/locationprovider/DefaultLocationProvider.java

```diff
@@ -271,19 +271,29 @@ public void runScheduledTask(@NonNull String taskId) {
         }
     }
 
-    @Override
+     @Override
     public void onPositiveButtonClick() {
-        boolean activityStarted = startActivityForResult(new Intent(Settings.ACTION_LOCATION_SOURCE_SETTINGS),
-              RequestCode.GPS_ENABLE);
-        if (!activityStarted) {
-            onLocationFailed(FailType.VIEW_NOT_REQUIRED_TYPE);
+        if (isGPSProviderEnabled()) {
+            LogUtils.logI("User didn't activate GPS through the dialog, but using status bar");
```

Comment:
Please update the text here and below, it doesn't actually necessary to do it via status bar, user can also directly go to settings and enabled it. Just mention as 'but activated manually'